tests/test_datasets/test_dataset.py

```diff
@@ -310,6 +322,93 @@ def ts_with_prediction_intervals(ts_without_target_components, prediction_interv
     return ts
 
 
+def test_create_ts_with_datetime_timestamp():
+    freq = "D"
+    df = generate_ar_df(periods=10, freq=freq, n_segments=3)
+    df_wide = TSDataset.to_dataset(df)
+    df_wide.index.freq = freq
+    ts = TSDataset(df=df_wide, freq=freq)
+
+    pd.testing.assert_index_equal(ts.index, df_wide.index)
+    pd.testing.assert_frame_equal(ts.to_pandas(), df_wide)
+
+
+def test_create_ts_with_int_timestamp():
+    df = generate_ar_df(periods=10, freq=None, n_segments=3)
+    df_wide = TSDataset.to_dataset(df)
+    ts = TSDataset(df=df_wide, freq=None)
+
+    pd.testing.assert_index_equal(ts.index, df_wide.index)
+    pd.testing.assert_frame_equal(ts.to_pandas(), df_wide)
+
+
+def test_create_ts_with_exog_datetime_timestamp():
+    freq = "D"
+    df = generate_ar_df(periods=10, start_time="2020-01-05", freq=freq, n_segments=3, random_seed=0)
+    df_exog = generate_ar_df(periods=20, start_time="2020-01-01", freq=freq, n_segments=3, random_seed=1)
+    df_exog.rename(columns={"target": "exog"}, inplace=True)
+
+    df_wide = TSDataset.to_dataset(df)
+    df_exog_wide = TSDataset.to_dataset(df_exog)
+    ts = TSDataset(df=df_wide, df_exog=df_exog_wide, freq=freq)
+
+    expected_merged = pd.concat([df_wide, df_exog_wide.loc[df_wide.index]], axis=1).sort_index(axis=1)
```

Comment:
In `TSDataset._merge_exog` we use `sort_index` with `level=(0, 1)`. May be here we also should use it?
```python
expected_merged = pd.concat([df_wide, df_exog_wide.loc[df_wide.index]], axis=1).sort_index(axis=1, level=(0, 1))
```